log/sequencer.go

```diff
@@ -360,18 +379,22 @@ func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, gu
 	// TODO(codingllama): Consider adding a source-aware replenish method
 	// (eg, qm.Replenish(ctx, tokens, specs, quota.SequencerSource)), so there's no ambiguity as to
 	// where the tokens come from.
-	if err := s.qm.PutTokens(ctx, len(leaves), []quota.Spec{
-		{Group: quota.Tree, Kind: quota.Read, TreeID: logID},
-		{Group: quota.Tree, Kind: quota.Write, TreeID: logID},
-		{Group: quota.Global, Kind: quota.Read},
-		{Group: quota.Global, Kind: quota.Write},
-	}); err != nil {
-		glog.Warningf("Failed to replenish tokens for tree %v: %v", logID, err)
+	if numLeaves > 0 {
+		tokens := int(float64(numLeaves) * quotaIncreaseFactor())
+		glog.V(2).Infof("Replenishing %v tokens for tree %v (numLeaves = %v)", tokens, logID, leaves)
+		if err := s.qm.PutTokens(ctx, tokens, []quota.Spec{
+			{Group: quota.Tree, Kind: quota.Read, TreeID: logID},
+			{Group: quota.Tree, Kind: quota.Write, TreeID: logID},
+			{Group: quota.Global, Kind: quota.Read},
+			{Group: quota.Global, Kind: quota.Write},
+		}); err != nil {
```

Comment:
Oops, sorry, I misunderstood you. That's a different approach from #741.

I'll go ahead and merge this, as I think we want it anyway. Let's take the discussion offline.